pkg/sql/compile/dml.go

```diff
@@ -40,31 +40,61 @@ func (s *Scope) Delete(c *Compile) (uint64, error) {
 	arg := s.Instructions[len(s.Instructions)-1].Arg.(*deletion.Argument)
 
 	var tableID string
+	var err error
+	var dbSource engine.Database
+	var rel engine.Relation
+	var isTemp bool
+
 	if arg.DeleteCtxs[0].CanTruncate {
-		dbSource, err := c.e.Database(c.ctx, arg.DeleteCtxs[0].DbName, c.proc.TxnOperator)
+		dbSource, err = c.e.Database(c.ctx, arg.DeleteCtxs[0].DbName, c.proc.TxnOperator)
 		if err != nil {
 			return 0, err
 		}
 
-		var rel engine.Relation
 		if rel, err = dbSource.Relation(c.ctx, arg.DeleteCtxs[0].TableName); err != nil {
-			return 0, err
+			var e error
+			dbSource, e = c.e.Database(c.ctx, engine.TEMPORARY_DBNAME, c.proc.TxnOperator)
+			if e != nil {
+				return 0, err
+			}
+			rel, e = dbSource.Relation(c.ctx, arg.DeleteCtxs[0].DbName+"-"+arg.DeleteCtxs[0].TableName)
+			if e != nil {
```

Comment:
sigh ...

Reply:
I stopped reading.   fix and resubmit.

Reply:
So for all these, do you agree the soulution to add a function to get them?

Reply:
@fengttt you said that encode the name, I try to use hash/crc32 to get a unique encoded string, do you argee?